oneflow/user/kernels/fft_kernels.cpp

```diff
@@ -0,0 +1,245 @@
+
+    if (input->data_type() == kFloat || input->data_type() == kDouble) {
+      FftR2CKernelUtil<device_type, dtype_in, dtype_out>::FftR2CForward(
+          ctx->stream(), input_ptr, out_ptr, input_shape, out_shape, input->stride(), out->stride(),
+          /*forward=*/true, dims, norm_fct, /*real_type=*/input->data_type());
+    } else {
+      CHECK_OR_THROW(false) << "expects kFloat or kDouble, but gets " << input->data_type();
+    }
+
+    if (!onesided) {
+      FillConjSymmetryUtil<device_type, dtype_out>::FillConjSymmetryForward(
+          ctx->stream(), out_ptr, out_shape, out->stride(), dims.back(), out_shape.elem_cnt());
+    }
+  }
+};
+
+template<DeviceType device_type, typename dtype_in, typename dtype_out>
+class FftC2RKernel final : public user_op::OpKernel {
+ public:
+  FftC2RKernel() = default;
+  ~FftC2RKernel() = default;
+
+ private:
+  bool AlwaysComputeWhenAllOutputsEmpty() const override { return false; }
+  void Compute(user_op::KernelComputeContext* ctx) const override {
+    const user_op::Tensor* input = ctx->Tensor4ArgNameAndIndex("input", 0);
+    user_op::Tensor* out = ctx->Tensor4ArgNameAndIndex("out", 0);
+    int64_t last_dim_size = ctx->Attr<int64_t>("last_dim_size");
+    double norm_fct = ctx->Attr<double>("norm_fct");
+    const std::vector<int64_t>& dims = ctx->Attr<std::vector<int64_t>>("dims");
+
+    const dtype_in* input_ptr = input->dptr<dtype_in>();
+    dtype_out* out_ptr = out->mut_dptr<dtype_out>();
+
+    Shape input_shape(input->shape_view());
+    Shape out_shape(out->shape_view());
+
+    out_shape[dims.back()] = last_dim_size;
+
+    if (input->data_type() == kComplex64 || input->data_type() == kComplex128) {
+      FftC2RKernelUtil<device_type, dtype_in, dtype_out>::FftC2RForward(
+          ctx->stream(), input_ptr, out_ptr, input_shape, out_shape, input->stride(), out->stride(),
+          /*forward=*/false,
+          /*last_dim_size=*/last_dim_size, dims, norm_fct, /*real_type=*/out->data_type());
+    } else {
+      CHECK_OR_THROW(false) << "expects kComplex64 or kComplex128, but gets " << input->data_type();
+    }
+  }
+};
+
+template<DeviceType device_type, typename dtype_in, typename dtype_out>
+class StftCpuKernel final : public user_op::OpKernel {
+ public:
+  StftCpuKernel() = default;
+  ~StftCpuKernel() = default;
+
+ private:
+  using user_op::OpKernel::Compute;
+  void Compute(user_op::KernelComputeContext* ctx) const override {
+    const user_op::Tensor* input = ctx->Tensor4ArgNameAndIndex("input", 0);
+    user_op::Tensor* output = ctx->Tensor4ArgNameAndIndex("output", 0);
+    user_op::Tensor* tmp_buffer = ctx->Tensor4ArgNameAndIndex("tmp_buffer", 0);
+    const auto normalized = ctx->Attr<bool>("normalized");
+    const auto return_complex = ctx->Attr<bool>("return_complex");
+    const bool onesided = ctx->Attr<bool>("onesided");
+
+    const ShapeView& input_shape = input->shape_view();
+    const ShapeView& output_shape = output->shape_view();
```

Comment:
view 类型一般只有两个指针的大小，用值传递性能会更好
```suggestion
    const ShapeView input_shape = input->shape_view();
    const ShapeView output_shape = output->shape_view();
```